pkg/authn/oidc.go

```diff
@@ -0,0 +1,40 @@
+package authn
+
+import (
+	"k8s.io/apiserver/pkg/authentication/authenticator"
+	"k8s.io/apiserver/pkg/authentication/request/bearertoken"
+	"k8s.io/apiserver/plugin/pkg/authenticator/token/oidc"
+)
+
+// OIDCConfig represents configuration used for JWT request authentication
+type OIDCConfig struct {
+	IssuerURL            string
+	ClientID             string
+	CAFile               string
+	UsernameClaim        string
+	UsernamePrefix       string
+	GroupsClaim          string
+	GroupsPrefix         string
+	SupportedSigningAlgs []string
+}
+
+// NewOIDCAuthenticator returns OIDC authenticator 
+func NewOIDCAuthenticator(authn *AuthnConfig) (authenticator.Request, error) {
```

Comment:
nit: just pass a `*OIDCConfig` here, we don't need the whole dependency on `*AuthnConfig`.